src/Infrastructure.Dapper/Vault/Repositories/SecurityTaskRepository.cs

```diff
@@ -1,18 +1,48 @@
-﻿using Bit.Core.Settings;
+﻿using System.Data;
+using Bit.Core.Settings;
 using Bit.Core.Vault.Entities;
+using Bit.Core.Vault.Enums;
 using Bit.Core.Vault.Repositories;
 using Bit.Infrastructure.Dapper.Repositories;
+using Dapper;
+using Microsoft.Data.SqlClient;
 
 namespace Bit.Infrastructure.Dapper.Vault.Repositories;
 
 public class SecurityTaskRepository : Repository<SecurityTask, Guid>, ISecurityTaskRepository
 {
     public SecurityTaskRepository(GlobalSettings globalSettings)
         : this(globalSettings.SqlServer.ConnectionString, globalSettings.SqlServer.ReadOnlyConnectionString)
-    { }
+    {
+    }
 
     public SecurityTaskRepository(string connectionString, string readOnlyConnectionString)
         : base(connectionString, readOnlyConnectionString)
-    { }
+    {
+    }
 
+    /// <inheritdoc />
+    public async Task<ICollection<SecurityTask>> GetManyByUserIdStatusAsync(Guid userId,
+        IEnumerable<SecurityTaskStatus> status = null)
+    {
+        var statusTable = new DataTable();
```

Comment:
🎨 If we go the route of making this a more generic `ByteEnumArray` we could add a new DapperHelper like `ToByteArrayTVP(this IEnumerable<byte> values)` to do this instead. [See here for more context.](https://github.com/bitwarden/server/blob/329eef82cdd68d1fa65bd883212bfcaaba7a325f/src/Infrastructure.Dapper/DapperHelpers.cs#L156-L176)